network/discovery/fork_listener.go

```diff
@@ -0,0 +1,160 @@
+package discovery
+
+import (
+	"sync"
+
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/ssvlabs/ssv/networkconfig"
+)
+
+// forkListener wraps a pre-fork and a post-fork listener.
+// Before the fork, it performs operations on both services.
+// Aftet the fork, it performs operations only on the post-fork service.
+type forkListener struct {
```

Comment:
since we already have a ForkListener in the node, we should probably rename this one so we don't have too many 😅 

maybe `forkingDV5Listener`?
(to parallel a similar mechanism we have called `forkingMessageValidation`)

Reply:
Agree, renaming